src/Icons/doc/index.rst

```diff
@@ -90,20 +158,127 @@ Usage
 HTML Syntax
 ~~~~~~~~~~~
 
+.. code-block:: html+twig
+
+    <twig:UX:Icon name="user-profile" />
+
+    {# Renders "user-profile.svg" #}
+    <twig:UX:Icon name="user-profile" class="w-4 h-4" />
+
+    {# Renders "sub-dir/user-profile.svg" (sub-directory) #}
+    <twig:UX:Icon name="sub-dir:user-profile" class="w-4 h-4" />
+
+    {# Renders "flowbite:user-solid" from ux.symfony.com #}
+    <twig:UX:Icon name="flowbite:user-solid" />
+
 .. note::
 
     ``symfony/ux-twig-component`` is required to use the HTML syntax.
 
-.. code-block:: html
+HTML Attributes
+~~~~~~~~~~~~~~~
+
+The second argument of the ``ux_icon`` function is an array of attributes to add to the icon.
+
+.. code-block:: twig
+
+    {# renders "user-profile.svg" with class="w-4 h-4" #}
+    {{ ux_icon('user-profile', {class: 'w-4 h-4'}) }}
+
+    {# renders "user-profile.svg" with class="w-4 h-4" and aria-hidden="true" #}
+    {{ ux_icon('user-profile', {class: 'w-4 h-4', 'aria-hidden': true}) }}
+
+Default Attributes
+~~~~~~~~~~~~~~~~~~
+
+You can set default attributes for all icons in your configuration. These attributes will be
+added to all icons unless overridden by the second argument of the ``ux_icon`` function.
+
+.. code-block:: yaml
+
+    # config/packages/ux_icons.yaml
+    ux_icons:
+        default_icon_attributes:
+            fill: currentColor
+
+Now, all icons will have the ``fill`` attribute set to ``currentColor`` by default.
+
+.. code-block:: twig
+
+    # renders "user-profile.svg" with fill="currentColor"
+    {{ ux_icon('user-profile') }}
+
+    # renders "user-profile.svg" with fill="red"
+    {{ ux_icon('user-profile', {fill: 'red'}) }}
+
+Accessibility
+-------------
 
-    <twig:UX:Icon name="user-profile" class="w-4 h-4" /> <!-- renders "user-profile.svg" -->
+Icons are a great way to add visual elements to your website, but they can be a challenge
+for accessibility.
 
-    <twig:UX:Icon name="sub-dir:user-profile" class="w-4 h-4" /> <!-- renders "sub-dir/user-profile.svg" (sub-directory) -->
+According to the `W3C <https://design-system.w3.org/styles/svg-icons.html>`_, there are
+three methods to improve icons accessibility, depending on the context:
 
-    <twig:UX:Icon name="flowbite:user-solid" /> <!-- renders "flowbite:user-solid" from ux.symfony.com -->
+    * **Decorative**: Icons that are purely decorative and do not convey any meaning or function.
+    * **Informative**: Icons that convey information or a function.
+    * **Functional**: Icons that are interactive and perform a function.
 
-Caching
--------
+Decorative Icons
```

Comment:
I'd made these sub-sections more concise by merging the initial description of each icon type that it's included at the beginning. I don't think we have to add subsections here. We can add a definition list like we do in many parts of Symfony Docs:

```rst
**Decorative icons**
    They are purely decorative and does not convey any meaning or function.
    Hide them from screen readers using the ``aria-hidden`` attribute:

    .. code-block:: html+twig

        <a href="/profile">
            {{ ux_icon('user-profile', {class: 'w-4 h-4', 'aria-hidden': true}) }}
            Back to profile
        </a>

**Informative icons**
    [...]

**Functional icons**
    [...]
```